doc/source/cluster/kubernetes/user-guides/upgrade-guide.md

```diff
@@ -26,54 +26,37 @@ If you want to understand the reasoning behind the CRD versioning plan, see [ray
-
-    # Install RayCluster v1.0.0 which uses CRD v1.
-    helm install raycluster kuberay/ray-cluster --version 1.0.0
-    # Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest:
-    # resource mapping not found for name: "raycluster-kuberay" namespace: "" from "": no
-    # matches for kind "RayCluster" in version "ray.io/v1"
-    # ensure CRDs are installed first
-    ```
-  * Example 2: Upgrade KubeRay from v0.6.0 to v1.0.0 with deleting the old CRD.
-    ```shell
-    # Install KubeRay v0.6.0 and CRD v1alpha1
-    helm install kuberay-operator kuberay/kuberay-operator --version 0.6.0
-
-    # The following instruction uninstalls only KubeRay v0.6.0. It does not uninstall CRD v1alpha1.
-    helm uninstall kuberay-operator
-
-    # Delete CRDs. Note that deleting the CRD causes a cascading deletion of custom resources.
-    kubectl delete crd rayclusters.ray.io
-    kubectl delete crd rayjobs.ray.io
-    kubectl delete crd rayservices.ray.io
-
-    # Install KubeRay v1.0.0 and new CRD including v1.
-    helm install kuberay-operator kuberay/kuberay-operator --version 1.0.0
-
-    # Check CRD
-    kubectl describe crd rayclusters.ray.io | grep v1
-    # You can see both "Name: v1alpha1" and "Name: v1".
-
-    # Install RayCluster v1.0.0 which uses CRD v1.
-    helm install raycluster kuberay/ray-cluster --version 1.0.0
-    ```
+Due to reliability and security implications of webhooks, KubeRay doesn't support a conversion webhook to convert v1alpha1 to v1 APIs.
+
+To upgrade the KubeRay version, follow these steps in order:
+1. Upgrade the CRD manifest, containing new fields added to the v1 CRDs.
+2. Upgrade the kuberay-operator image to the new version.
+3. Verify the success of the upgrade.
+
+The following is an example of upgrading KubeRay from v1.0.0 to v1.1.0:
+```
+# Upgrade the CRD to v1.1.0.
+# Note: This example uses kubectl because Helm doesn't support lifecycle management of CRDs.
+# See the Helm documentation for more details: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations
+$ kubectl replace -k "github.com/ray-project/kuberay/ray-operator/config/crd?ref=v1.1.0"
+
+# Upgrade kuberay-operator to v1.1.0. This step doesn't upgrade the CRDs.
+$ helm upgrade kuberay-operator kuberay/kuberay-operator --version v1.1.0
+
+# Install a RayCluster using the v1.1.0 helm chart to verify the success of the upgrade.
+$ helm install raycluster kuberay/ray-cluster --version 1.1.0
+```
+
+The following is an example of upgrading KubeRay from v0.6.0 to v1.0.0:
+```
+# Upgrade the CRD to v1.0.0.
+# Note: This example uses kubectl because Helm doesn't support lifecycle management of CRDs.
+# See the Helm documentation for more details: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations
+$ kubectl replace -k "github.com/ray-project/kuberay/ray-operator/config/crd?ref=v1.0.0"
```

Comment:
FYI this step causes disruptions for any running `RayService` resources until you apply the `RayService` config again. See this slack thread for more details: https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1701370178891199

Reply:
Is that specific to  upgrading the CRD first only or are you referring to the entire v0.6.0 to v1.0.0 upgrade? As far as I'm aware the 0.6.0 -> 1.0.0 is known to be disruptive and breaking changes like that are expected but I'm not super familiar with RayService changes from 0.6.0 and 1.0.0 yet to know

Reply:
cc @andrewsykim do you plan to address this comment in this PR? If you want to do it in a follow-up PR, I am comfortable merging this PR.

Reply:
If you're planning to add the 0.6.0 -> 1.0.0 guidance in a separate PR, I'd suggest changing the example here to show upgrade from 1.0.0 -> 1.1.0 or something like that otherwise it could be misleading

Reply:
@smit-kiri are you interested in opening a PR to share your experience after this PR got merged?

Reply:
The old guidance suggests deleting the CRD, so I think what we have in this PR is still better 

I don't have enough experience with 0.6.0 to 1.0.0 upgrade issues right now so it'll have to be a follow up PR if we add one.

Reply:
@kevin85421 Yeah I can open a PR!

Reply:
Thanks, @smit-kiri!